src/test/java/uk/nhs/hee/tis/revalidation/connection/controller/ConnectionControllerTest.java

```diff
@@ -81,6 +107,41 @@ public void shouldRemoveDoctor() throws Exception {
         .andExpect(status().isOk());
   }
 
+  @Test
+  public void shouldReturnAllConnectionsForADoctor() throws Exception {
+    final var connectionDto = prepareConnectionDto();
+    when(connectionService.getTraineeConnectionInfo(gmcId)).thenReturn(connectionDto);
+    this.mockMvc.perform(get("/api/connections/{gmcId}", gmcId))
+        .andExpect(status().isOk())
+        .andDo(print())
+        .andExpect(content().json(objectMapper.writeValueAsString(connectionDto)));
+  }
+
+  @Test
+  public void shouldNotFailWhenThereIsNoConnectionsForADoctor() throws Exception {
+    when(connectionService.getTraineeConnectionInfo(gmcId)).thenReturn(new ConnectionDto());
+    this.mockMvc.perform(get("/api/connections/{gmcId}", gmcId))
+        .andExpect(status().isOk())
+        .andDo(print());
```

Comment:
If this is being used for manual verification of the result, please try to replace it with automated assertions.

Reply:
It is updated, please have a look. Thanks